File: src/main/java/com/velocity/velocity_management/teammonthlycalculation/entity/TeamVelocity.java
package com.velocity.velocity_management.teammonthlycalculation.entity;

import com.velocity.velocity_management.team.entity.Team;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(
        name = "team_velocities",
        uniqueConstraints = {
                @UniqueConstraint(
                        name = "uk_team_velocity_period",
                        columnNames = {"team_id", "year", "month"}
                )
        })
public class TeamVelocity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "team_id", nullable = false)
    private Team team;

    @Column(nullable = false)
    private Integer year;

    @Column(nullable = false)
    private Integer month;

    @Column(nullable = false)
    private Double teamVelocityRatio;

    @Column(nullable = false)
    private Integer totalMembers;

    @Column(nullable = false)
    private Integer validatedMembers;

    @Column(nullable = false)
    private Integer unvalidatedMembers;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;
}

File: frontend/src/app/core/models/team-velocity.ts
export type TeamVelocityStatus = 'PENDING_VALIDATION' | 'VALIDATED' | 'NOT_CALCULATED';

export interface CollaboratorVelocitySummary {
  collaboratorId: number;
  collaboratorName: string;
  velocityRatio: number | null;
  status: TeamVelocityStatus;
  includedInCalculation: boolean;
}

export interface TeamVelocity {
  id: number;
  teamId: number;
  teamName: string;
  year: number;
  month: number;
  teamVelocityRatio: number;
  totalMembers: number;
  validatedMembers: number;
  unvalidatedMembers: number;
  createdAt: string;
  updatedAt: string;
}

export interface TeamVelocityDetail extends TeamVelocity {
  members: CollaboratorVelocitySummary[];
}

export interface CreateTeamVelocityRequest {
  teamId: number;
  year: number;
  month: number;
}
